### ISSUE TITLE
Problem with ctrl+C signal (does not stop the command)

### ISSUE THREAD

**Issue by Wirone** [PHP-CS-Fixer/PHP-CS-Fixer#7783]
We currently have problem with ctrl+C signal (does not stop the command when run in Docker). The `fix` / `check` commands must stop gracefully when SIGTERM is invoked.

```shell
$ docker run --rm -it -v $(pwd):/code ghcr.io/php-cs-fixer/php-cs-fixer:3-php8.3 check
```

and then ctrl+C should stop the command.

_Originally posted by @Wirone in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7782#issuecomment-1912011174_

**Comment by Wirone** [comment]
It looks like it moving `pcntl` to the lower layer and including it in the Docker image does not solve the problem anyway, but running Docker container with `--init` makes ctrl+C work properly 🤔. The change in #7782 is correct anyway, because `symfony/console` requires `pcntl` to work with signals.

**Comment by Wirone** [comment]
@hgiesenow do you know what can be the root cause of Fixer not reacting to ctrl+C when running with `docker run`? When I run Fixer inside container and use ctrl+C inside the container, the Fixer's process stops properly.

```
$ docker run --rm -it --init -v $(pwd):/code --entrypoint="" fixer:local sh
/code # php-cs-fixer check
PHP CS Fixer 3.48.1-DEV Small Changes by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 8.3.2
Loaded config default from "/code/.php-cs-fixer.php".
    0/1070 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%^C
/code #
```

`fixer:local` image used above is a local build (`docker build --target dist -t fixer:local .`) on branch with correct `pcntl`.

**Comment by Wirone** [comment]
While working on #8021 I've just tested and ctrl+C works properly:

```
$ docker run -it --rm -v $(pwd):/code ghcr.io/php-cs-fixer/php-cs-fixer:3.57-php8.3 check

WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
PHP CS Fixer 3.57.1 7th Gear by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 8.3.7
Running analysis on 7 cores with 10 files per process.
Parallel runner is an experimental feature and may be unstable, use it at your own risk. Feedback highly appreciated!
Loaded config default from "/code/.php-cs-fixer.php".
Using cache file "/code/.php-cs-fixer.cache.json".
  336/1118 [▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░]  30%^C
Found 0 of 1118 files that can be fixed in 3.379 seconds, 16.000 MB memory used
```

I'll close this, we can reopen later if needed.
